File: C++ Programming/Competitive Programming/Coding Challenges 2/program5.cpp
/*
Q5. Accept number of rows and number of columns from user and display below
pattern.
Input : iRow = 3 iCol = 4
Output :1   2   3   4
        5   6   7   8
        9   10  11  12
*/

#include<iostream>
using namespace std;

void Pattern(int iRow, int iCol)
{
    int i=0;
    int j=0;
    int iCnt = 1;
    
cout<<"Output is: \n";

    for ( i = 1; i <= iRow; i++)
    {
        for ( j = 1; j <= iCol; j++)
        {
            cout<<iCnt<<"\t";
            iCnt++;
        }
        cout<<"\n";
    }
}

int main()
{
int iValue1 = 0, iValue2 = 0;

cout<<"Enter Number of Rows : ";
cin>>iValue1;

cout<<"Enter Number of Columns : ";
cin>>iValue2;

Pattern(iValue1, iValue2);

return 0;
}

File: Java Programming/Competitive Programming/Coding Challenges 7/program5.java
/*
5. Accept number of rows and number of columns from user and display below pattern.
Input : iRow = 4 iCol = 4
Output :1 2 3 4
          2 3 4
            3 4
              4
*/

import java.util.*;

class program5
{
    public static void Pattern(int iRow, int iCol)
    {
        for (int i = 1; i <= iRow; i++)
        {
            int sum = i - 1;

            for (int j = 1; j <= sum; j++)
            {
                System.out.print(" \t");
            }

            for (int j = i; j <= iCol; j++)
            {
                System.out.print(j+"\t");
            }

            System.out.println("");
        }
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0;
        int iValue2 = 0;

        System.out.println("Enter number of rows :");
        iValue1 = sobj.nextInt();

        System.out.println("Enter number of columns :");
        iValue2 = sobj.nextInt();

        Pattern(iValue1, iValue2);

        sobj.close();
    }
}